### PR TITLE
[misc] simplify the smoke test and process shutdown

### PR DESCRIPTION
**Pull request by das7pad**
### Description

For https://github.com/overleaf/issues/issues/2782#issuecomment-625269923

This PR removes the dependency on the out-dated `smoke-test-sharelatex` package and replaces the test running via mocha with simple requests and non-raising assertions.

_Basically a port of https://github.com/das7pad/clsi-sharelatex/commit/fe83b4451da0b4408208b9337e088ef3d24c304e_ 

#### Related Issues / PRs

https://github.com/overleaf/issues/issues/2782#issuecomment-625269923

### Review

I preserved the logic of not caching test results from `/smoke_test_force` runs, but I am happy to change that (and simplify `test/smoke/js/SmokeTests.js` even further).

As part of the `resCacher`  refactoring I changed the process shutdown to a `setTimeout` handler which manipulates a variable in the parent scope -- there is no need to check for the process age on every health check run.

#### Potential Impact

High. CLSI vms could cycle frequently due to failing probes.

#### Manual Testing Performed

- enable smoke tests in the settings defaults (`smokeTest`) and start clsi in the dev-env
- `curl -i 127.0.0.1:3013/health_check` returns `OK` (200)
- `curl -i 127.0.0.1:3013/smoke_test_force` returns `OK` (200)
- restarting the container and immediately running `curl -i 127.0.0.1:3013/health_check` returns `SmokeTestsPending` (500)
- break the compiles: change docker image in settings defaults to a non texlive image (`clsi.docker.image`) and restart clsi
- `curl -i 127.0.0.1:3013/health_check` returns `no pdf returned` (500)
- `curl -i 127.0.0.1:3013/smoke_test_force` returns `no pdf returned` (500)

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -192,64 +192,39 @@ app.get('/oops', function(req, res, next) {
...
-  },
-
-  // default the server to be down
-  code: 500,
-  body: {},
-  setContentType: 'application/json'
-}
-
-let shutdownTime
+let PROCESS_IS_TOO_OLD = false
```
Comment (briangough) [inline app.js:195]: I'd put this on Settings.processIsTooOld. We have been putting these shutdown related flags on Settings so they can be checked in other parts of the code when needed..

File: app.js [hunk 2]
```diff
@@ -192,64 +192,39 @@ app.get('/oops', function(req, res, next) {
...
-      logger.log('running smoke tests')
-      smokeTest.run(
-        require.resolve(__dirname + '/test/smoke/js/SmokeTests.js')
-      )({}, resCacher)
-      return setTimeout(runSmokeTest, 30 * 1000)
+  function runSmokeTest() {
+    if (PROCESS_IS_TOO_OLD) return
+    logger.log('running smoke tests')
+    smokeTest.triggerRun(err => {
+      logger.error({ err }, 'smoke tests failed')
```
Comment (briangough) [inline app.js:215]: This gets logged every time, maybe make it conditional on `if(err)`.

**Comment by briangough** [comment]
Otherwise no problems!

File: app.js (hunk 2 above)
Comment (das7pad) [reply app.js:215]: Right, I remember to having fixed that -- https://github.com/das7pad/clsi-sharelatex/commit/23e52cf7b76a7b1195e89760edde24c74259f872

**Pull request merged**